### PR TITLE
backupccl: remove targets from restore resumer

### PR DESCRIPTION
**Pull request by pbardea**
This commit removes stored targets from the restore resumer. The resumer
should only be used to store a reference to a job and should not be used
to store state.

This commit also fixes a bug where a RESTORE would fail if run during a
cluster upgrade from 20.1 to 20.2 where the table descriptors, which
were stored on the descriptors, would be created on every resumption of
the restore job. The bug would manifest as:

- Start the restore job on a 20.1 node, which creates table descriptors
with UnexposedParentID as 0 (when read from a 20.2 node)
- Pause the job
- Upgrade the cluster
- Resume the job on a 20.2 node
- The restore job recreates the in-memory representation of the
descriptors with UnexposedParentID now as 29.
- When the job completes, the job checks that the table descriptor has
not changed, but the in-memory descriptor is faulty.

This is now fixed since we no longer store the descriptor in the
resumer, and always refer to the descriptor in the job details.

Release justification: bug fix
Release note (bug fix): Fix bug when running restore while upgrading the
cluster from 20.1 to 20.2.

### CODE REVIEW

File: pkg/jobs/jobspb/jobs.proto [hunk 1]
```diff
@@ -153,15 +157,21 @@ message RestoreDetails {
     bytes info = 2;
   }
   repeated Tenant tenants = 13 [(gogoproto.nullable) = false];
+
   string override_db = 6 [(gogoproto.customname) = "OverrideDB"];
+
+  // The restore job has several atomic stages. For now, we keep track of which
+  // stages have completed via these flags. It would be nice if this was handled
```
Comment (dt) [inline pkg/jobs/jobspb/jobs.proto:164]: ... I'm not sure what jobs would handle here? surely it shouldn't know about the states within a restore job?

**Review by dt — APPROVED** [review]

File: pkg/jobs/jobspb/jobs.proto (hunk 1 above)
Comment (pbardea) [reply pkg/jobs/jobspb/jobs.proto:164]: The comment may have been a bit too wishful. I was thinking that a job could be defined as a sequence of stages, each with their own corresponding rollback step if required. Haven't thought too deeply about it, but I'll update the comment. Better tracked elsewhere anyway.

**Comment by pbardea** [comment]
Running into another bug while adding some tests for this - but it wasn't introduced by this change. Will address in a separate PR.

**Comment by pbardea** [comment]
TFTR!
bors r=dt

**Comment by lucy-zhang** [comment]
cc @ajwerner

**Pull request merged**